internal/shell/shell.go

```diff
@@ -137,6 +137,10 @@ type shellAutoCompleter struct {
 }
 
 func (sac *shellAutoCompleter) Do(line []rune, pos int) (newLine [][]rune, lengh int) {
+	// 32 is the rune repr for " "
+	if line[len(line)-1] == 32 {
+		return nil, 0
+	}
 	suggestions := sac.suggestCompletion(string(line[0:pos]))
```

Comment:
A few comments:

1. Remember that not only `' '` is a space. `unicode.IsSpace` checks for all kind of spaces (like tabs)
2. Instead of using the code for checking only `' '`, you could use `== ' '`
3. See that at line 144 we don't pass the entire string to suggester, but `line[0:pos]`. That's for the cases where the cursor isn't at the last position of the input. Something like `sele<cursor here> * from table;`

```suggestion
	if unicode.IsSpace(line[len(pos)-1]) {
		return nil, 0
	}
	suggestions := sac.suggestCompletion(string(line[0:pos]))
```